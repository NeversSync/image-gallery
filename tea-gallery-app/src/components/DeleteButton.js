import React from 'react';
import PropTypes from 'prop-types';

DeleteButton.propTypes = {
  images: PropTypes.array,
};

export default function DeleteButton(props) {
  return (
    <button onClick={props.onDeleteImage}>Delete</button>
  );
}