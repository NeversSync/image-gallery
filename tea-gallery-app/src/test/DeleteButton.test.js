import React from 'react';
import renderer from 'react-test-renderer';
import DeleteButton from '../components/DeleteButton';

const images = [
  {
    title: 'tea leaves in teapot',
    description: 'what nice leaves',
    url: '/../public/img/teainpot.jpg'
  },
  {
    title: 'Tetsubin over charcoal',
    description: 'what a nice tetsubin',
    url: '/../public/img/tetsubin-charcoal.jpg'
  },
  {
    title: 'Teapot display',
    description: 'what a nice teapot display',
    url: '/../public/img/teapot-display.jpg'
  }
];

describe('DeleteButton', () => {
  it('renders the DeleteButton component', () => {
    const component = renderer.create(<DeleteButton images={images}/>);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });

});