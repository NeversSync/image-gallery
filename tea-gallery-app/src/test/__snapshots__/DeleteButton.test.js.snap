// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteButton renders the DeleteButton component 1`] = `
<button
  onClick={undefined}
>
  Delete
</button>
`;
